crates/types/src/nodes_config.rs

```diff
@@ -500,6 +500,8 @@ pub enum MetadataServerState {
     /// The server is an active member of the metadata store cluster.
     #[default]
     Member,
+    /// The server should try to automatically join a metadata store cluster.
+    AutoJoin,
```

Comment:
So to draw an analogy to the rest of the per-role states,  `AutoJoin => Provisioning`, `Member => Active`.  `Standby => Disabled`. Got it :) 

Perhaps we can have aliases to these or rename them at some point for consistency?